In [1]:
# Define the ticket prices for full-fare and economy class
price_full_fare = 440
price_economy = 218

# Calculate the critical fractile
critical_fractile = (price_full_fare - price_economy) / price_full_fare

# Print the critical fractile
print(f"Critical Fractile: {critical_fractile}")

# Given empirical distribution table
# This should ideally come from the empirical data provided
# For this example, let's assume it's an array of tuples with demand and cumulative probability
empirical_distribution = [
    (40, 0.25), (41, 0.31), (42, 0.35), 
    (43, 0.36), (44, 0.42), (45, 0.49),
    (46, 0.51), (47, 0.54), (48, 0.57), 
    (49, 0.62), (50, 0.65), (51, 0.70),
    (52, 0.74), (53, 0.80), (54, 0.89), 
    (55, 1.00)
]

# Find the smallest cumulative probability that is greater than or equal to the critical fractile
protection_level = next(demand for demand, cum_prob in empirical_distribution if cum_prob >= critical_fractile)

# Total number of coach seats
total_seats = 230

# Calculate the booking limit for economy seats
booking_limit_economy = total_seats - protection_level

# Print the protection level and booking limit for economy seats
print(f"Protection Level for Full-fare Seats (Q*): {protection_level}")
print(f"Booking Limit for Economy Seats (BL): {booking_limit_economy}")

Critical Fractile: 0.5045454545454545
Protection Level for Full-fare Seats (Q*): 46
Booking Limit for Economy Seats (BL): 184


# Airline Revenue Management Analysis

## Protection Level and Booking Limit Calculation

### Problem Statement
An airline offers two fare classes for coach seats on a particular flight: full-fare class at \$440/ticket and economy class at \$218/ticket. There are 230 coach seats on the aircraft. The demand for full-fare seats has an empirical distribution with a mean of 43 and a standard deviation of 8.

### a) Calculation of Protection Level and Booking Limit for Low-fare Seats

1. **Critical Fractile Calculation**:
   The critical fractile is calculated using the formula:
   
   $q = F^{-1}\left(\frac{p-c}{p}\right)$
   
   Where $p$ is the full-fare price, $c$ is the economy fare price, and $F^{-1}$ represents the inverse of the cumulative distribution function of the demand.
   
   Given that $p = \$440$ and $c = \$218$, we compute the critical fractile as:
   
   $Critical\ Fractile = \frac{440 - 218}{440} \approx 0.505$
   
2. **Protection Level**:
   By comparing the critical fractile with the empirical distribution, we determine the protection level $Q^*$ as the full-fare demand $Q$ at the smallest cumulative probability greater than or equal to the critical fractile.
   
   Protection Level (Q^*): `46`

3. **Booking Limit for Economy Seats**:
   The booking limit for economy seats (BL) is calculated as the total number of coach seats minus the protection level.
   
   Booking Limit (BL): `184`

### b) Impact of Last-minute Discounted Sales on Protection Level

- **Expectation**: The protection level $Q^*$ is expected to be **Higher**.
- **Justification**: The opportunity to sell unsold seats at the last minute at a reduced rate reduces the opportunity cost of reserving a seat for a full-fare customer. This can potentially increase the protection level since the risk of a seat remaining unsold is mitigated by the possibility of last-minute sales.
